### PR TITLE
#29 Removed CountDownLatch, added support for test suite

### PR DESCRIPTION
**Pull request by vvolverine**

### CODE REVIEW

File: sprimber-engine/src/main/java/com/griddynamics/qa/sprimber/engine/model/TestSuite.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ */
+
+public class TestSuite {
+
+    // TODO: 2019-06-08 Implement the logic for population before/after suite hooks
+    private List<ActionDefinition> hooks = new ArrayList<>();
+    private List<TestCase> testCases = new ArrayList<>();
+    private String runtimeId;
+
+    public List<ActionDefinition> getHooks() {
```
Comment (L-Evg) [inline sprimber-engine/src/main/java/com/griddynamics/qa/sprimber/engine/model/TestSuite.java]: Lombok everywhere to decrease code line count?

File: sprimber-engine/src/main/java/com/griddynamics/qa/sprimber/engine/model/configuration/SprimberProperties.java [hunk 2]
```diff
@@ -53,4 +57,66 @@ public void setFeaturePath(String featurePath) {
...
+        this.tsExecutorPool = tsExecutorPool;
+    }
+
+    public static class PoolExecutorProperties {
+        private int coreSize;
+        private int maxSIze;
+        private int keepAliveSeconds;
+        private String threadNamePrefix;
+
+        public PoolExecutorProperties(int coreSize, int maxSIze, int keepAliveSeconds, String threadNamePrefix) {
```
Comment (L-Evg) [inline sprimber-engine/src/main/java/com/griddynamics/qa/sprimber/engine/model/configuration/SprimberProperties.java]: maxSIze -> maxSize probably.

File: sprimber-engine/src/main/java/com/griddynamics/qa/sprimber/engine/model/configuration/SprimberProperties.java [hunk 3]
```diff
@@ -53,4 +57,66 @@ public void setFeaturePath(String featurePath) {
...
+        return tsExecutorPool;
+    }
+
+    public void setTsExecutorPool(PoolExecutorProperties tsExecutorPool) {
+        this.tsExecutorPool = tsExecutorPool;
+    }
+
+    public static class PoolExecutorProperties {
+        private int coreSize;
+        private int maxSIze;
```
Comment (L-Evg) [inline sprimber-engine/src/main/java/com/griddynamics/qa/sprimber/engine/model/configuration/SprimberProperties.java]: maxSIze -> maxSize probably.

File: sprimber-engine/src/main/java/com/griddynamics/qa/sprimber/engine/model/configuration/SprimberProperties.java [hunk 4]
```diff
@@ -37,6 +37,10 @@ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 
     private List<String> tagFilters = new ArrayList<>();
     private String featurePath;
+    private PoolExecutorProperties tcExecutorPool =
+            new PoolExecutorProperties(3, 3, 60, "TC-Executor-");
```
Comment (L-Evg) [inline sprimber-engine/src/main/java/com/griddynamics/qa/sprimber/engine/model/configuration/SprimberProperties.java]: These numbers  in the code look really confusing. What about moving them to properties?

File: sprimber-engine/src/main/java/com/griddynamics/qa/sprimber/engine/processor/cucumber/CucumberSuiteProvider.java [hunk 5]
```diff
@@ -78,25 +80,29 @@ public CucumberProcessor(Parser<GherkinDocument> gherkinParser,
...
         try {
-            testCases = Arrays.stream(applicationContext.getResources(sprimberProperties.getFeaturePath()))
-                    .map(this::buildCucumberDocument)
+            List<TestCase> testCases = Arrays.stream(applicationContext.getResources(sprimberProperties.getFeaturePath()))
+                    .map(resource -> buildCucumberDocument(resource, suiteId))
                     .flatMap(this::buildTestCasesStream)
                     .collect(Collectors.toList());
+            testSuite.addTestCasesForExecution(testCases);
         } catch (IOException e) {
             e.printStackTrace();
```
Comment (L-Evg) [inline sprimber-engine/src/main/java/com/griddynamics/qa/sprimber/engine/processor/cucumber/CucumberSuiteProvider.java]: a logger?

File: sprimber-engine/src/main/java/com/griddynamics/qa/sprimber/engine/model/TestSuite.java (hunk 1 above)
Comment (L-Evg) [reply sprimber-engine/src/main/java/com/griddynamics/qa/sprimber/engine/model/TestSuite.java]: Sure, the comment was mainly about "one day in for future".

File: sprimber-engine/src/main/java/com/griddynamics/qa/sprimber/engine/model/configuration/SprimberProperties.java (hunk 4 above)
Comment (L-Evg) [reply sprimber-engine/src/main/java/com/griddynamics/qa/sprimber/engine/model/configuration/SprimberProperties.java]: I would prefer to have that default properties file - a single entry point for all the default values.

**Comment by vvolverine** [comment]
No longer relevant, similar and better functionality implemented as part of 1.1.0

**Pull request closed without merging**